valqueries-test/src/test/java/com/valqueries/automapper/SqlGeneratorITBase.java

```diff
@@ -79,21 +83,55 @@ public void setup() {
 	}
 
 	protected abstract DataSource getDataSource();
+	protected abstract DataSource secondaryDatasource();
+
+	@Test
+	public void generateTable_withProvidedDatabase() {
+		// arrange
+		final DataSource otherDataSource = secondaryDatasource();
+		final Database databaseToUse = new Database(otherDataSource);
+		final Database databaseToIgnore = database;
+
+		update("DROP TABLE IF EXISTS "+dialect.getTableName(Clazz.of(SimpleTestTable.class)));
+		update("DROP TABLE IF EXISTS "+dialect.getTableName(Clazz.of(SimpleTestTable.class)), databaseToUse);
+
+		// act
+		sqlGenerator.generateCreateTable(TypeDescriberImpl.getTypeDescriber(SimpleTestTable.class), otherDataSource);
+
+		// assert
+		SqlDescriber.DbTable tableThatShouldBeCreated = describeSimpleTestTable(databaseToUse);
+		SqlDescriber.DbTable tableThatShouldNotBeCreated = describeSimpleTestTable(databaseToIgnore);
+
+		assertEquals(3, tableThatShouldBeCreated.columns.size());
+		assertNotNull(tableThatShouldBeCreated.columns.get("id"));
+		assertNotNull(tableThatShouldBeCreated.columns.get("title").getType());
+		assertNotNull(tableThatShouldBeCreated.columns.get("created_at").getType());
```

Comment:
You are checking that the first column exists but that the other two columns have a type, why the difference?